batch-processing/helpers.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"image"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"errors"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/cenkalti/backoff/v4"
+)
+
+const (
+	InvalidCSVFormat   = "csv file must have a header row with 'url' as the first column"
+	CouldNotFetchImage = "received status %d when trying to download image"
+	EmptyCSV           = "provided CSV file appears to be empty"
+)
+
+/**
+* Download the image from the URL
+* @param: {string} URL - the URL of the image to download
+* @return: {io.Reader} body - the body of the image
+* @return: {string} ext - the file extension (format) of the image
+* @return: {error} err - any error that occurred
+ */
+func DownloadFileFromUrl(URL string) (io.Reader, string, [16]byte, error) {
```

Comment:
Looks like you're missing documentation for the `[16]byte` that's returned?

Reply:
Good catch!